s3/pkg/meta/db/drivers/tidb/bucket.go

```diff
@@ -225,39 +243,47 @@ func (t *TidbClient) ListObjects(ctx context.Context, bucketName, marker, verIdM
 	objectMap := make(map[string]struct{})
 	objectNum := make(map[string]int)
 	commonPrefixes := make(map[string]struct{})
-	omarker := marker
+	omarker := filter[common.KMarker]
+	delimiter := filter[common.KDelimiter]
+	prefix := filter[common.KPrefix]
 	for {
 		var loopcount int
 		var sqltext string
 		var rows *sql.Rows
-		if marker == "" {
-			sqltext = "select bucketname,name,version,nullversion,deletemarker from objects where bucketName=? " +
-				"order by bucketname,name,version limit ?;"
-			rows, err = t.Client.Query(sqltext, bucketName, maxKeys)
-		} else {
-			sqltext = "select bucketname,name,version,nullversion,deletemarker from objects where bucketName=? " +
-				"and name >=? order by bucketname,name,version limit ?,?;"
-			rows, err = t.Client.Query(sqltext, bucketName, marker, objectNum[marker], objectNum[marker]+maxKeys)
+		args := make([]interface{}, 0)
+		// only select index column here to avoid slow query
+		sqltext = "select bucketname,name,version from objects where bucketName=?"
+		args = append(args, bucketName)
+		var inargs []interface{}
+		sqltext, inargs, err = buildSql(ctx, filter, sqltext)
+		for _, v := range inargs {
+			args = append(args, v)
 		}
+		log.Infof("sqltext:%s, args:%v\n", sqltext, args)
+		tstart := time.Now()
+		rows, err = t.Client.Query(sqltext, args...)
 		if err != nil {
 			err = handleDBError(err)
 			return
 		}
+		tqueryend := time.Now()
+		tdur := tqueryend.Sub(tstart).Nanoseconds()
+		if tdur/1000000 > TimeDur {
```

Comment:
what is the rational here ?

Reply:
if the intent is to call out that the query is slow, imho, better we add
- the actual time the query took
- the number of rows 
- if we can, the query explain plan at this point in time ? 
the idea is to add information that will help us find out WHY the query was slow...
and how do we define what slow means ? How to quantify it ?

Reply:
The actual time the query took is print in log.
And if we add too many information in code to help us to find out why the query was slow, then that will make the normal query be slower. 
So we just want to print a log if the query is slow, and if that happened, then we need to connect to database to check the number of rows and other information to find out why.